cdap-event-reader-spi/src/main/java/io/cdap/cdap/spi/events/EventReader.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.spi.events;
+
+
+/**
+ * {@link EventReader} Interface for listening for events.
+ */
+public interface EventReader<T extends Event> extends AutoCloseable {
+  /**
+   * Method to initialize EventReader.
+   *
+   * @param context Configuration properties of reader
+   */
+  void initialize(EventReaderContext context);
+
+  /**
+   * Pull messages if available.
```

Comment:
Is this method blocking on having at least one message? Or it can return empty result? It's better to be specific about the behavior of this method.

Reply:
The returned EventResult can be empty, however the PubSub pull implementation will block for a non-trivial amount of time before admitting that there are no messages available. What do you think of the updated JavaDoc?